app/controllers/foreman_bootdisk/api/v2/subnet_disks_controller.rb

```diff
@@ -42,6 +43,10 @@ def subnet
         def resource_scope
           Subnet.authorized('view_subnets')
         end
+
+        def subnet_not_found
+          not_found ("Subnet not found by id '%s'") % params[:id]
```

Comment:
Subnet with id %s not found

sounds probably little bit better. Tests fail tho.

Reply:
I'd still prefer a cleaner message, but whatever.